File: app/src/main/java/com/example/ui/screens/SplashScreen.kt
package com.example.ui.screens

import androidx.compose.animation.core.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.delay

@Composable
fun SplashScreen(onSplashComplete: () -> Unit) {
  // Navigation trigger
  LaunchedEffect(Unit) {
    delay(2500L)
    onSplashComplete()
  }

  Box(
    modifier = Modifier
      .fillMaxSize()
      .background(Color(0xFF0F172A)),
    contentAlignment = Alignment.Center
  ) {
    Column(
      horizontalAlignment = Alignment.CenterHorizontally,
      verticalArrangement = Arrangement.Center
    ) {
      // SVG Logo Representation: Rounded box with #1a1a2e fill
      Box(
        modifier = Modifier
          .size(120.dp)
          .clip(RoundedCornerShape(28.dp))
          .background(Color(0xFF1A1A2E)),
        contentAlignment = Alignment.Center
      ) {
        // Gradient text "JR" matching the SVG linearGradient from #7c3aed to #3b82f6
        Text(
          text = "JR",
          style = TextStyle(
            fontFamily = FontFamily.SansSerif,
            fontWeight = FontWeight.Bold,
            fontSize = 54.sp,
            brush = Brush.linearGradient(
              colors = listOf(Color(0xFF7C3AED), Color(0xFF3B82F6))
            )
          )
        )
      }

      Spacer(modifier = Modifier.height(24.dp))

      // Logo Text Label
      Text(
        text = "JobsReport",
        fontWeight = FontWeight.Black,
        fontSize = 24.sp,
        color = Color.White
      )
      
      Text(
        text = "SECURE INTELLIGENCE PORTAL",
        fontFamily = FontFamily.Monospace,
        fontSize = 11.sp,
        color = Color(0xFF10B981),
        fontWeight = FontWeight.Bold,
        modifier = Modifier.padding(top = 4.dp)
      )

      Spacer(modifier = Modifier.height(56.dp))

      // Bouncing dots loader resembling Facebook loading dots
      FacebookLoadingDots()

      Spacer(modifier = Modifier.height(24.dp))

      Text(
        text = "CONNECTING TERMINAL...",
        fontFamily = FontFamily.Monospace,
        fontSize = 10.sp,
        color = Color(0xFF64748B)
      )
    }
  }
}

@Composable
fun FacebookLoadingDots() {
  val dots = remember { listOf(Animatable(0f), Animatable(0f), Animatable(0f), Animatable(0f)) }

  dots.forEachIndexed { index, animatable ->
    LaunchedEffect(animatable) {
      // Stagger start of each dot animation to create wave bounce effect
      delay(index * 150L)
      while (true) {
        // Bounce up
        animatable.animateTo(
          targetValue = 1f,
          animationSpec = tween(
            durationMillis = 350,
            easing = FastOutSlowInEasing
          )
        )
        // Bounce down
        animatable.animateTo(
          targetValue = 0f,
          animationSpec = tween(
            durationMillis = 350,
            easing = FastOutSlowInEasing
          )
        )
        // Stagger rest between cycles
        delay(200L)
      }
    }
  }

  Row(
    horizontalArrangement = Arrangement.spacedBy(10.dp),
    verticalAlignment = Alignment.CenterVertically
  ) {
    dots.forEach { animatable ->
      val size = 12.dp
      // Translate Y based on animation value
      val translationY = -(animatable.value * 14.dp.value)
      
      Box(
        modifier = Modifier
          .size(size)
          .offset(y = translationY.dp)
          .clip(CircleShape)
          .background(
            Brush.linearGradient(
              colors = listOf(Color(0xFF7C3AED), Color(0xFF3B82F6))
            )
          )
      )
    }
  }
}

File: lib/screens/splash_screen.dart
import 'dart:async';
import 'package:flutter/material.dart';

class SplashScreen extends StatefulWidget {
  const SplashScreen({super.key});

  @override
  State<SplashScreen> createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen> with SingleTickerProviderStateMixin {
  late AnimationController _controller;
  late List<Animation<double>> _animations;

  @override
  void initState() {
    super.initState();

    // 2.5s Splash Delay
    Timer(const Duration(milliseconds: 2500), () {
      Navigator.pushReplacementNamed(context, '/home');
    });

    // Animation for loading dots
    _controller = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 1200),
    )..repeat();

    _animations = List.generate(3, (index) {
      final start = index * 0.2;
      final end = start + 0.6;
      return Tween<double>(begin: 0.0, end: 1.0).animate(
        CurvedAnimation(
          parent: _controller,
          curve: Interval(start, end, curve: Curves.easeInOut),
        ),
      );
    });
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF0F172A),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            // JR Gradient Logo Card
            Container(
              width: 120,
              height: 120,
              decoration: BoxDecoration(
                color: const Color(0xFF1A1A2E),
                borderRadius: BorderRadius.circular(28),
              ),
              alignment: Alignment.center,
              child: ShaderMask(
                shaderCallback: (bounds) => const LinearGradient(
                  colors: [Color(0xFF7C3AED), Color(0xFF3B82F6)],
                  begin: Alignment.topLeft,
                  end: Alignment.bottomRight,
                ).createShader(bounds),
                child: const Text(
                  'JR',
                  style: TextStyle(
                    color: Colors.white,
                    fontWeight: FontWeight.bold,
                    fontSize: 54,
                  ),
                ),
              ),
            ),
            const SizedBox(height: 24),

            // Logo Label
            const Text(
              'JobsReport',
              style: TextStyle(
                fontWeight: FontWeight.black,
                fontSize: 24,
                color: Colors.white,
              ),
            ),
            const SizedBox(height: 4),
            const Text(
              'SECURE INTELLIGENCE PORTAL',
              style: TextStyle(
                fontFamily: 'monospace',
                fontSize: 11,
                color: Color(0xFF10B981),
                fontWeight: FontWeight.bold,
              ),
            ),
            const SizedBox(height: 56),

            // Animated Loading Dots
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: List.generate(3, (index) {
                return AnimatedBuilder(
                  animation: _controller,
                  builder: (context, child) {
                    final double val = _animations[index].value;
                    final double dy = -10.0 * (val > 0.5 ? (1.0 - val) : val) * 2;
                    return Transform.translate(
                      offset: Offset(0, dy),
                      child: Container(
                        width: 8,
                        height: 8,
                        margin: const EdgeInsets.symmetric(horizontal: 4),
                        decoration: const BoxDecoration(
                          color: Color(0xFF10B981),
                          shape: BoxShape.circle,
                        ),
                      ),
                    );
                  },
                );
              }),
            ),
            const SizedBox(height: 24),

            const Text(
              'CONNECTING TERMINAL...',
              style: TextStyle(
                fontFamily: 'monospace',
                fontSize: 10,
                color: Color(0xFF64748B),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
